bundles/CoreBundle/EventListener/AssetSanitizationListener.php

```diff
@@ -0,0 +1,83 @@
+<?php
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ * @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ * @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\EventListener;
+
+
+use Pimcore\Event\AssetEvents;
+use Pimcore\Event\Model\ElementEventInterface;
+use Pimcore\Model\Asset;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use enshrined\svgSanitize\Sanitizer;
+use Symfony\Component\Mime\MimeTypes;
+
+/**
+ * @internal
+ */
+class AssetSanitizationListener implements EventSubscriberInterface
+{
+    public static function getSubscribedEvents()
+    {
+        return [
+            AssetEvents::PRE_ADD => 'sanitizeAsset',
+            AssetEvents::PRE_UPDATE => 'sanitizeAsset',
+        ];
+    }
+
+    /**
+     * @param ElementEventInterface $e
+     */
+    public function sanitizeAsset(ElementEventInterface $e)
+    {
+        $element = $e->getElement();
+
+        if ($element instanceof Asset &&
+            $element->getType() === 'image' &&
+            $element->getDataChanged()
```

Comment:
Is `$dataChanged` also `true` for `PRE_ADD` events?

Reply:
yes, i confirm this, it is changed to `true` everytime `setStream` is called and on creation of a new Asset it is initiated [there](https://github.com/pimcore/pimcore/blob/73142e5464f60f91313ad06aeff618de721a6df0/models/Asset.php#L380) and then [set](https://github.com/pimcore/pimcore/blob/73142e5464f60f91313ad06aeff618de721a6df0/models/Asset.php#L398)